### DISCUSSION TITLE
Getting the last dobj using dependency matcher.

### DISCUSSION THREAD

**Discussion by rishkum** [explosion/spaCy#7789] [Help: Coding & Implementations] [answered]
Hi I am trying to get the last dobj using dependency matcher. 

s1 = """We are giving new ways of getting credit"""

    pattern_product_offering =     [ 
    {
    "RIGHT_ID": "anchore_Offering",
    "RIGHT_ATTRS": {"ORTH": {"IN": ["offering", "providing", "giving"]}}
    },
    {
    "LEFT_ID": "anchore_Offering",
    "REL_OP": ">",
    "RIGHT_ID": "who_offering",
    "RIGHT_ATTRS": {"DEP": "nsubj"},
    },
    {
    "LEFT_ID": "anchore_Offering",
    "REL_OP": ">>",
    "RIGHT_ID": "What_is_being_offered",
    "RIGHT_ATTRS": {"DEP": "dobj"},
    },

    {
    "LEFT_ID": "What_is_being_offered",
    "REL_OP": ">",
    "RIGHT_ID": "founded_object_modifier",
    "RIGHT_ATTRS": {"DEP": {"IN": ["compound","pcomp", "amod"]}},
    }]

```
def findMatchingSentencess(s1):
    matcher = DependencyMatcher(nlp.vocab)
   
    # pattern: "[subject] ... initially founded"

    matcher.add("offering", [pattern_product_offering])

    doc = nlp(self.statement)
    matches = matcher(doc)
    if len(matches) > 0:
        match_id, token_ids = matches[0]
        for i in range(len(token_ids)):
            print(TextDependency.pattern_product_offering[i]["RIGHT_ID"] + ":", doc[token_ids[i]].text)
    else:
        return("NO MATCH")
```
I am getting outputs like this:
anchore_Offering: giving
who_offering: We
What_is_being_offered: ways
founded_object_modifier: new

instead of 
anchore_Offering: giving
who_offering: We
What_is_being_offered: **credit**
founded_object_modifier: new

credit is the last dobj in the sentence,s1, is there a way to target this? [OP](https://spacy.io/api/matcher) isn't applicable for this.

spacy 3.0.1
Model used : tried both md and lg

**Comment by adrianeboyd** [comment, marked as answer]
Hi, the problem is that the modifier is required by the pattern and the final dobj doesn't have a modifier. Unfortunately the dependency matcher doesn't currently support optional nodes in its patterns.
